qiskit_ibm/runtime/utils.py

```diff
@@ -145,7 +145,9 @@ def default(self, obj: Any) -> Any:  # pylint: disable=arguments-differ
             return {'__type__': 'settings',
                     '__module__': obj.__class__.__module__,
                     '__class__': obj.__class__.__name__,
-                    '__value__': obj.settings}
+                    '__value__': {
+                        **obj.settings,
+                        'fidelity': obj.fidelity} if hasattr(obj, 'fidelity') else obj.settings}
```

Comment:
You don't need this anymore as the fix is now in qiskit-terra

Reply:
I removed this block from `utils.py`, but it seems `fidelity` is still not being serialized in the settings? The prop still needs to be popped from the input settings before the `assert` for the serialization test to pass.

Reply:
Functions cannot be serialized. So if you want test a `QNSPSA` with fidelity it needs to be a non callable.